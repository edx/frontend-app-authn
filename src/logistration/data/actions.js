import { AsyncActionType } from '../../data/utils';

export const REGISTER_NEW_USER = new AsyncActionType('REGISTRATION', 'REGISTER_NEW_USER');
export const LOGIN_REQUEST = new AsyncActionType('LOGIN', 'REQUEST');
export const THIRD_PARTY_AUTH_CONTEXT = new AsyncActionType('THIRD_PARTY_AUTH', 'GET_THIRD_PARTY_AUTH_CONTEXT');

// Register

export const registerNewUser = registrationInfo => ({
  type: REGISTER_NEW_USER.BASE,
  payload: { registrationInfo },
});

export const registerNewUserBegin = () => ({
  type: REGISTER_NEW_USER.BEGIN,
});

export const registerNewUserSuccess = (redirectUrl, success) => ({
  type: REGISTER_NEW_USER.SUCCESS,
  payload: { redirectUrl, success },
});

export const registerNewUserFailure = () => ({
  type: REGISTER_NEW_USER.FAILURE,
});

// Login
export const loginRequest = creds => ({
  type: LOGIN_REQUEST.BASE,
  payload: { creds },
});

export const loginRequestBegin = () => ({
  type: LOGIN_REQUEST.BEGIN,
});

export const loginRequestSuccess = (redirectUrl, success) => ({
  type: LOGIN_REQUEST.SUCCESS,
  payload: { redirectUrl, success },
});

export const loginRequestFailure = () => ({
  type: LOGIN_REQUEST.FAILURE,
});

// Third party auth context
export const getThirdPartyAuthContext = (urlParams) => ({
  type: THIRD_PARTY_AUTH_CONTEXT.BASE,
  payload: { urlParams },
});

export const getThirdPartyAuthContextBegin = () => ({
  type: THIRD_PARTY_AUTH_CONTEXT.BEGIN,
});

export const getThirdPartyAuthContextSuccess = (thirdPartyAuthContext) => ({
  type: THIRD_PARTY_AUTH_CONTEXT.SUCCESS,
  payload: { thirdPartyAuthContext },
});

export const getThirdPartyAuthContextFailure = () => ({
  type: THIRD_PARTY_AUTH_CONTEXT.FAILURE,
});
