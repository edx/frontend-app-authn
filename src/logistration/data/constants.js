export const INACTIVE_USER = 'inactive-user';
export const INTERNAL_SERVER_ERROR = 'internal-server-error';
export const NON_COMPLIANT_PASSWORD_EXCEPTION = 'NonCompliantPasswordException';
