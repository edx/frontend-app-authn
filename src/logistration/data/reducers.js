import {
  REGISTER_NEW_USER,
  LOGIN_REQUEST,
  THIRD_PARTY_AUTH_CONTEXT,
} from './actions';

export const defaultState = {
  registrationResult: {},
  loginResult: {},
};

const reducer = (state = defaultState, action) => {
  switch (action.type) {
    case REGISTER_NEW_USER.BEGIN:
      return {
        ...state,
      };
    case REGISTER_NEW_USER.SUCCESS:
      return {
        ...state,
        registrationResult: action.payload,
      };
    case REGISTER_NEW_USER.FAILURE:
      return {
        ...state,
      };
    case LOGIN_REQUEST.BEGIN:
      return {
        ...state,
      };
    case LOGIN_REQUEST.SUCCESS:
      return {
        ...state,
        loginResult: action.payload,
      };
    case LOGIN_REQUEST.FAILURE:
      return {
        ...state,
      };
    case THIRD_PARTY_AUTH_CONTEXT.BEGIN:
      return {
        ...state,
      };
    case THIRD_PARTY_AUTH_CONTEXT.SUCCESS:
      return {
        ...state,
        thirdPartyAuthContext: action.payload.thirdPartyAuthContext,
      };
    case THIRD_PARTY_AUTH_CONTEXT.FAILURE:
      return {
        ...state,
      };
    default:
      return state;
  }
};

export default reducer;
