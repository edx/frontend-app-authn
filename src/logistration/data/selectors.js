import { createSelector } from 'reselect';

export const storeName = 'logistration';

export const logistrationSelector = state => ({ ...state[storeName] });

export const loginRequestSelector = createSelector(
  logistrationSelector,
  logistration => logistration.loginResult,
);

export const loginErrorSelector = createSelector(
  logistrationSelector,
  logistration => logistration.loginError,
);

export const registrationRequestSelector = createSelector(
  logistrationSelector,
  logistration => logistration.registrationResult,
);

export const thirdPartyAuthContextSelector = createSelector(
  logistrationSelector,
  logistration => logistration.thirdPartyAuthContext,
);
