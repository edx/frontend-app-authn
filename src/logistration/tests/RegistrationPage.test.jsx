import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { getConfig } from '@edx/frontend-platform';
import { IntlProvider, injectIntl, configure } from '@edx/frontend-platform/i18n';

import RegistrationPage from '../RegistrationPage';
import { RenderInstitutionButton } from '../InstitutionLogistration';
import { PENDING_STATE } from '../../data/constants';
import { fetchRegistrationForm, fetchRealtimeValidations, registerNewUser } from '../data/actions';

const IntlRegistrationPage = injectIntl(RegistrationPage);
const mockStore = configureStore();

describe('./RegistrationPage.js', () => {
  const initialState = {
    logistration: {
      registrationResult: { success: false, redirectUrl: '' },
      thirdPartyAuthContext: {
        platformName: 'openedX',
        currentProvider: null,
        finishAuthUrl: null,
        providers: [],
        secondaryProviders: [],
      },
      registrationError: null,
      formData: {
        fields: [{
          label: 'I agree to the Your Platform Name Here <a href="/honor" rel="noopener" target="_blank">Honor Code</a>',
          name: 'honor_code',
          type: 'checkbox',
          errorMessages: {
            required: 'You must agree to the Your Platform Name Here Honor Code',
          },
          required: true,
        }],
      },
    },
  };

  let props = {};
  let store = {};

  const appleProvider = {
    id: 'oa2-apple-id',
    name: 'Apple',
    iconClass: null,
    iconImage: 'https://edx.devstack.lms/logo.png',
    loginUrl: '/auth/login/apple-id/?auth_entry=login&next=/dashboard',
  };

  const secondaryProviders = {
    id: 'saml-test',
    name: 'Test University',
    loginUrl: '/dummy-auth',
    registerUrl: '/dummy_auth',
  };

  const reduxWrapper = children => (
    <IntlProvider locale="en">
      <Provider store={store}>{children}</Provider>
    </IntlProvider>
  );

  beforeEach(() => {
    store = mockStore(initialState);
    configure({
      loggingService: { logError: jest.fn() },
      config: {
        ENVIRONMENT: 'production',
        LANGUAGE_PREFERENCE_COOKIE_NAME: 'yum',
      },
      messages: {
        'es-419': {},
        de: {},
        'en-us': {},
      },
    });
    props = {
      registrationResult: jest.fn(),
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should show error messages on invalid extra fields', () => {
    const validationMessage = {
      honorCode: 'You must agree to the Your Platform Name Here Honor Code',
      country: 'Select your country or region of residence.',
    };
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        formData: {
          fields: [
            ...initialState.logistration.formData.fields,
            {
              label: 'The country or region where you live.',
              name: 'country',
              type: 'select',
              options: [{ value: '', name: '--' }, { value: 'AF', name: 'Afghanistan' }],
              errorMessages: {
                required: validationMessage.country,
              },
              required: true,
            },
          ],
        },
      },
    });
    const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));

    registrationPage.find('input#honor_code').simulate('change', { target: { checked: false, name: 'honor_code', type: 'checkbox' } });
    registrationPage.update();
    expect(registrationPage.find('#honor_code-invalid-feedback').text()).toEqual(validationMessage.honorCode);

    registrationPage.find('select#country').simulate('change', { target: { checked: false, name: 'country', type: 'checkbox' } });
    registrationPage.update();
    expect(registrationPage.find('#country-invalid-feedback').text()).toEqual(validationMessage.country);
  });

  it('should toggle optional fields state on checkbox click', () => {
    const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));

    registrationPage.find('input#optional').simulate('change', { target: { checked: true } });
    expect(registrationPage.find('RegistrationPage').state('enableOptionalField')).toEqual(true);
  });

  it('should toggle optional fields state on text click', () => {
    const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));

    registrationPage.find('#additionalFields').simulate('click');
    expect(registrationPage.find('RegistrationPage').state('enableOptionalField')).toEqual(true);
  });

  it('should show optional fields section on optional check enabled', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        formData: {
          fields: [
            {
              label: 'Tell us why you\'re interested in edX',
              name: 'goals',
              type: 'textarea',
              required: false,
            },
            {
              label: 'Highest level of Education completed.',
              name: 'level_of_education',
              type: 'select',
              options: [{ value: '', name: '--' }, { value: 'p', name: 'Doctorate' }],
              required: false,
            },
          ],
        },
      },
    });

    const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));

    registrationPage.find('input#optional').simulate('change', { target: { checked: true } });
    registrationPage.update();
    expect(registrationPage.find('textarea#goals').length).toEqual(1);
    expect(registrationPage.find('select#level_of_education').length).toEqual(1);
  });

  it('should dispatch fetchRegistrationForm on ComponentDidMount', () => {
    store = mockStore({
      ...initialState,
    });

    store.dispatch = jest.fn(store.dispatch);
    mount(reduxWrapper(<IntlRegistrationPage {...props} />));
    expect(store.dispatch).toHaveBeenCalledWith(fetchRegistrationForm());
  });

  it('should dispatch fetchRealtimeValidations on Blur', () => {
    const formPayload = {
      email: '',
      username: '',
      password: '',
      name: '',
      honor_code: true,
      country: '',
    };
    store.dispatch = jest.fn(store.dispatch);

    const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));

    registrationPage.find('input#username').simulate('blur', { target: { value: '', name: 'username' } });
    expect(store.dispatch).toHaveBeenCalledWith(fetchRealtimeValidations(formPayload));

    registrationPage.find('input#name').simulate('blur', { target: { value: '', name: 'name' } });
    expect(store.dispatch).toHaveBeenCalledWith(fetchRealtimeValidations(formPayload));

    registrationPage.find('input#email').simulate('blur', { target: { value: '', name: 'email' } });
    expect(store.dispatch).toHaveBeenCalledWith(fetchRealtimeValidations(formPayload));

    registrationPage.find('input#password').simulate('blur', { target: { value: '', name: 'password' } });
    expect(store.dispatch).toHaveBeenCalledWith(fetchRealtimeValidations(formPayload));
  });

  it('should not dispatch registerNewUser on Submit', () => {
    const formPayload = {
      email: '',
      username: '',
      password: '',
      name: '',
      honor_code: true,
      country: '',
    };
    store.dispatch = jest.fn(store.dispatch);

    const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
    registrationPage.find('button.submit').simulate('click');
    expect(store.dispatch).not.toHaveBeenCalledWith(registerNewUser(formPayload));
  });

  it('should match default section snapshot', () => {
    const tree = renderer.create(reduxWrapper(<IntlRegistrationPage {...props} />));
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match pending button state snapshot', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        submitState: PENDING_STATE,
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlRegistrationPage {...props} />));
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match TPA provider snapshot', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          providers: [appleProvider],
        },
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlRegistrationPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display no password field when current provider is present', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          ...initialState.logistration.thirdPartyAuthContext,
          currentProvider: 'Google',
        },
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlRegistrationPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('tests shouldComponentUpdate with pipeline user data', () => {
    const nextProps = {
      validations: null,
      thirdPartyAuthContext: {
        pipelineUserDetails: {
          name: 'test',
          email: 'test@example.com',
          username: 'test-username',
        },
      },
    };

    const root = mount(reduxWrapper(<IntlRegistrationPage {...props} />));

    const shouldUpdate = root.find('RegistrationPage').instance().shouldComponentUpdate(nextProps);
    expect(shouldUpdate).toBe(false);
  });

  it('should match url after redirection', () => {
    const dasboardUrl = 'http://test.com/testing-dashboard/';
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        registrationResult: {
          success: true,
          redirectUrl: dasboardUrl,
        },
      },
    });
    delete window.location;
    window.location = { href: '' };
    renderer.create(reduxWrapper(<IntlRegistrationPage />));
    expect(window.location.href).toBe(dasboardUrl);
  });

  it('should display institution register button', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          ...initialState.logistration.thirdPartyAuthContext,
          secondaryProviders: [secondaryProviders],
        },
      },
    });
    const root = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
    expect(root.text().includes('Use my institution/campus credentials')).toBe(true);
  });

  it('should not display institution register button', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          ...initialState.logistration.thirdPartyAuthContext,
          secondaryProviders: [secondaryProviders],
        },
      },
    });
    const root = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
    root.find(RenderInstitutionButton).simulate('click', { institutionLogin: true });
    expect(root.text().includes('Test University')).toBe(true);
  });

  it('should match url after TPA redirection', () => {
    const authCompleteUrl = '/auth/complete/google-oauth2/';
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        registrationResult: {
          success: true,
          redirectUrl: '',
        },
        thirdPartyAuthContext: {
          ...initialState.logistration.thirdPartyAuthContext,
          finishAuthUrl: authCompleteUrl,
        },
      },
    });

    delete window.location;
    window.location = { href: '' };

    renderer.create(reduxWrapper(<IntlRegistrationPage {...props} />));
    expect(window.location.href).toBe(getConfig().LMS_BASE_URL + authCompleteUrl);
  });

  it('should redirect to social auth provider url', () => {
    const registerUrl = '/auth/login/apple-id/?auth_entry=register&next=/dashboard';
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          providers: [{
            ...appleProvider,
            registerUrl,
          }],
        },
        formData: {
          fields: [{
            label: 'I agree to the Your Platform Name Here <a href="/honor" rel="noopener" target="_blank">Honor Code</a>',
            name: 'honor_code',
            type: 'checkbox',
            errorMessages: {
              required: 'You must agree to the Your Platform Name Here Honor Code',
            },
          }],
        },
      },
    });

    delete window.location;
    window.location = { href: '' };

    const loginPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));

    loginPage.find('button#oa2-apple-id').simulate('click');
    expect(window.location.href).toBe(getConfig().LMS_BASE_URL + registerUrl);
  });

  it('should match third party auth alert', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          ...initialState.logistration.thirdPartyAuthContext,
          currentProvider: 'Apple',
        },
      },
    });

    const expectedMessage = 'You\'ve successfully signed into Apple. We just need a little more information before '
                            + 'you start learning with openedX.';

    const loginPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
    expect(loginPage.find('#tpa-alert').find('span').text()).toEqual(expectedMessage);
  });

  it('should show error message on 409', () => {
    const windowSpy = jest.spyOn(global, 'window', 'get');
    windowSpy.mockImplementation(() => ({
      scrollTo: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    }));

    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        registrationError: {
          email: [
            {
              user_message: 'It looks like test@gmail.com belongs to an existing account. Try again with a different email address.',
            },
          ],
          username: [
            {
              user_message: 'It looks like test belongs to an existing account. Try again with a different username.',
            },
          ],
        },
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlRegistrationPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
    windowSpy.mockClear();
  });
});
