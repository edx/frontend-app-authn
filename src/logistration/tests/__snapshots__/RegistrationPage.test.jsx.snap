// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./RegistrationPage.js should display no password field when current provider is present 1`] = `
<div
  className="register-container mx-auto"
>
  <div
    className="fade alert-warning mt-n2 alert show"
    id="tpa-alert"
    role="alert"
  >
    <span>
      You've successfully signed into Google. We just need a little more information before you start learning with openedX.
    </span>
  </div>
  <div
    className="text-left"
  >
    <span>
      Already have an edX account?
    </span>
    <a
      href="/login"
    >
       Sign in.
    </a>
  </div>
  <form
    className="mb-4 mx-auto form-group"
  >
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="name"
      >
        Full Name (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="username"
      >
        Public Username (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="email"
      >
        Email (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={true}
        className="form-check-input"
        id="honor_code"
        name="honor_code"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={true}
      />
      I agree to the Your Platform Name Here 
      <a
        href="/honor"
        onClick={[Function]}
        target="_self"
      >
        Honor Code
      </a>
      
      <strong
        className="invalid-feedback"
        id="honor_code-invalid-feedback"
      >
        You must agree to the Your Platform Name Here Honor Code
      </strong>
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={false}
        className="form-check-input"
        id="optional"
        name="optional"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={false}
      />
      <p
        id="additionalFields"
        onClick={[Function]}
        role="presentation"
      >
        Support education research by providing additional information
      </p>
    </div>
    <button
      aria-disabled={false}
      aria-live="assertive"
      className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit mt-4 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="d-flex align-items-center justify-content-center"
      >
        Create Account
      </span>
    </button>
  </form>
</div>
`;

exports[`./RegistrationPage.js should match TPA provider snapshot 1`] = `
<div
  className="register-container mx-auto"
>
  <div
    className="text-left"
  >
    <span>
      Already have an edX account?
    </span>
    <a
      href="/login"
    >
       Sign in.
    </a>
  </div>
  <div
    className="d-block mb-4 mt-4"
  >
    <span
      className="d-block mx-auto mb-4 section-heading-line"
    >
      Create an account using
    </span>
    <div
      className="row tpa-container"
    >
      <button
        className="btn-social btn-oa2-apple-id"
        id="oa2-apple-id"
        onClick={[Function]}
        type="button"
      >
        <div
          aria-hidden="true"
        >
          <img
            alt="icon Apple"
            className="icon-image"
            src="https://edx.devstack.lms/logo.png"
          />
        </div>
        <span
          aria-hidden="true"
          className="pl-2"
        >
          Apple
        </span>
      </button>
    </div>
    <span
      className="d-block mx-auto text-center mt-4 section-heading-line"
    >
      or create a new one here
    </span>
  </div>
  <form
    className="mb-4 mx-auto form-group"
  >
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="name"
      >
        Full Name (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="username"
      >
        Public Username (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="email"
      >
        Email (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="password"
      >
        Password (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={true}
        className="form-check-input"
        id="honor_code"
        name="honor_code"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={true}
      />
      I agree to the Your Platform Name Here 
      <a
        href="/honor"
        onClick={[Function]}
        target="_self"
      >
        Honor Code
      </a>
      
      <strong
        className="invalid-feedback"
        id="honor_code-invalid-feedback"
      >
        You must agree to the Your Platform Name Here Honor Code
      </strong>
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={false}
        className="form-check-input"
        id="optional"
        name="optional"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={false}
      />
      <p
        id="additionalFields"
        onClick={[Function]}
        role="presentation"
      >
        Support education research by providing additional information
      </p>
    </div>
    <button
      aria-disabled={false}
      aria-live="assertive"
      className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit mt-4 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="d-flex align-items-center justify-content-center"
      >
        Create Account
      </span>
    </button>
  </form>
</div>
`;

exports[`./RegistrationPage.js should match default section snapshot 1`] = `
<div
  className="register-container mx-auto"
>
  <div
    className="text-left"
  >
    <span>
      Already have an edX account?
    </span>
    <a
      href="/login"
    >
       Sign in.
    </a>
  </div>
  <form
    className="mb-4 mx-auto form-group"
  >
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="name"
      >
        Full Name (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="username"
      >
        Public Username (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="email"
      >
        Email (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="password"
      >
        Password (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={true}
        className="form-check-input"
        id="honor_code"
        name="honor_code"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={true}
      />
      I agree to the Your Platform Name Here 
      <a
        href="/honor"
        onClick={[Function]}
        target="_self"
      >
        Honor Code
      </a>
      
      <strong
        className="invalid-feedback"
        id="honor_code-invalid-feedback"
      >
        You must agree to the Your Platform Name Here Honor Code
      </strong>
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={false}
        className="form-check-input"
        id="optional"
        name="optional"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={false}
      />
      <p
        id="additionalFields"
        onClick={[Function]}
        role="presentation"
      >
        Support education research by providing additional information
      </p>
    </div>
    <button
      aria-disabled={false}
      aria-live="assertive"
      className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit mt-4 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="d-flex align-items-center justify-content-center"
      >
        Create Account
      </span>
    </button>
  </form>
</div>
`;

exports[`./RegistrationPage.js should match pending button state snapshot 1`] = `
<div
  className="register-container mx-auto"
>
  <div
    className="text-left"
  >
    <span>
      Already have an edX account?
    </span>
    <a
      href="/login"
    >
       Sign in.
    </a>
  </div>
  <form
    className="mb-4 mx-auto form-group"
  >
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="name"
      >
        Full Name (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="username"
      >
        Public Username (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="email"
      >
        Email (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="password"
      >
        Password (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={true}
        className="form-check-input"
        id="honor_code"
        name="honor_code"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={true}
      />
      I agree to the Your Platform Name Here 
      <a
        href="/honor"
        onClick={[Function]}
        target="_self"
      >
        Honor Code
      </a>
      
      <strong
        className="invalid-feedback"
        id="honor_code-invalid-feedback"
      >
        You must agree to the Your Platform Name Here Honor Code
      </strong>
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={false}
        className="form-check-input"
        id="optional"
        name="optional"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={false}
      />
      <p
        id="additionalFields"
        onClick={[Function]}
        role="presentation"
      >
        Support education research by providing additional information
      </p>
    </div>
    <button
      aria-disabled={true}
      aria-live="assertive"
      className="pgn__stateful-btn pgn__stateful-btn-state-pending btn-primary submit mt-4 disabled btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="d-flex align-items-center justify-content-center"
      >
        <span
          className="pgn__stateful-btn-icon"
        >
          <span
            aria-hidden={true}
            className="icon fa fa-spinner fa-spin"
            id="Icon1"
          />
        </span>
        Create Account
      </span>
    </button>
  </form>
</div>
`;

exports[`./RegistrationPage.js should show error message on 409 1`] = `
<div
  className="register-container mx-auto"
>
  <div
    className="fade alert alert-danger show"
    role="alert"
  >
    <h4
      style={
        Object {
          "color": "#a0050e",
        }
      }
    >
      <span>
        We couldn't create your account.
      </span>
    </h4>
    <div>
      <ul>
        <li
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          It looks like test@gmail.com belongs to an existing account. Try again with a different email address.
        </li>
        <li
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          It looks like test belongs to an existing account. Try again with a different username.
        </li>
      </ul>
    </div>
  </div>
  <div
    className="text-left"
  >
    <span>
      Already have an edX account?
    </span>
    <a
      href="/login"
    >
       Sign in.
    </a>
  </div>
  <form
    className="mb-4 mx-auto form-group"
  >
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="name"
      >
        Full Name (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="username"
      >
        Public Username (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="email"
      >
        Email (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="h6 pt-3"
        htmlFor="password"
      >
        Password (required)
      </label>
      <input
        aria-describedby=""
        className="form-control"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={true}
        className="form-check-input"
        id="honor_code"
        name="honor_code"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={true}
      />
      I agree to the Your Platform Name Here 
      <a
        href="/honor"
        onClick={[Function]}
        target="_self"
      >
        Honor Code
      </a>
      
      <strong
        className="invalid-feedback"
        id="honor_code-invalid-feedback"
      >
        You must agree to the Your Platform Name Here Honor Code
      </strong>
    </div>
    <div
      className="form-group custom-control"
    >
      <input
        aria-describedby=""
        checked={false}
        className="form-check-input"
        id="optional"
        name="optional"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={false}
      />
      <p
        id="additionalFields"
        onClick={[Function]}
        role="presentation"
      >
        Support education research by providing additional information
      </p>
    </div>
    <button
      aria-disabled={false}
      aria-live="assertive"
      className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit mt-4 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="d-flex align-items-center justify-content-center"
      >
        Create Account
      </span>
    </button>
  </form>
</div>
`;
